packages/flutter_tools/lib/src/commands/ide_config.dart

```diff
@@ -0,0 +1,221 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+import '../template.dart';
+
+class IdeConfigCommand extends FlutterCommand {
+  IdeConfigCommand() {
+    argParser.addFlag(
+      'overwrite',
+      negatable: true,
+      defaultsTo: false,
+      help: 'When performing operations, overwrite existing files.',
+    );
+    argParser.addFlag(
+      'update-templates',
+      negatable: false,
+      help: 'Update the templates in the template directory from the current '
+          'configuration files. This is the opposite of what $name usually does. '
+          'Will search the flutter tree for .iml files and copy any missing ones '
+          'into the template directory. If --overwrite is also specified, it will '
+          'update any out-of-date files, and remove any deleted files from the '
+          'template directory.',
+    );
+  }
+
+  @override
+  final String name = 'ide-config';
+
+  @override
+  final String description = 'Configure the IDE for use in the Flutter tree.\n\n'
+      'If run on a Flutter tree that is already configured for the IDE, this '
+      'command will add any new configurations, recreate any files that are '
+      'missing. If --overwrite is specified, will revert existing files to '
+      'the template versions, reset the module list, and return configuration '
+      'settings to the template versions.\n\n'
+      'This command is intended for Flutter developers to help them set up the'
+      "Flutter tree for development in an IDE. It doesn't affect other projects.\n\n"
+      'Currently, IntelliJ is the default (and only) IDE that may be configured.';
+
+  @override
+  String get invocation => '${runner.executableName} $name';
+
+  static const String _ideName = 'intellij';
+  Directory get _templateDirectory => templateDirectoryInPackage(_ideName);
+  Directory get _flutterRoot => fs.directory(fs.path.absolute(Cache.flutterRoot));
+
+  // Returns true if any entire path element is equal to dir.
+  bool _hasDirectoryInPath(FileSystemEntity entity, String dir) {
+    String path = entity.absolute.path;
+    while (path.isNotEmpty && fs.path.dirname(path) != path) {
+      if (fs.path.basename(path) == dir) {
+        return true;
+      }
```

Comment:
nit: avoidable braces

>Avoid braces around one-line code blocks (e.g. in if statements) unless there's a chain (with else) and one of the elements in the chain has a multi-line block.

This nit applies at several places in the PR.

Reply:
Actually, we recently relaxed this, the style guide only says that single-line bodies that wrap to more than one line need to have braces: the "avoid" section has been removed.  The Dart style guide is explicitly the other way, with "DO use curly braces for all flow control structures."

Reply:
Ok, thanks for the info.

I suppose [this linter issue](https://github.com/dart-lang/linter/issues/142) should be updated to remove what I quoted.